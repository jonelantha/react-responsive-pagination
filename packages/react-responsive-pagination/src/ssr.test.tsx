import Pagination from './index.js';
import React from 'react';
import ReactDOMServer from 'react-dom/server';

describe('ssr', () => {
  test('renders the narrowest nav invisibly', () => {
    const component = <Pagination current={1} total={5} onPageChange={() => {}} />;

    const ssrComponent = ReactDOMServer.renderToString(component);

    expect(ssrComponent).toMatchSnapshot();
  });

  test('renders without console errors or warnings', () => {
    const mockConsoleError = jest.spyOn(console, 'error');
    const mockConsoleWarn = jest.spyOn(console, 'warn');

    try {
      const component = <Pagination current={1} total={5} onPageChange={() => {}} />;

      ReactDOMServer.renderToString(component);

      expect(mockConsoleError).not.toHaveBeenCalled();
      expect(mockConsoleWarn).not.toHaveBeenCalled();
    } finally {
      mockConsoleError.mockRestore();
      mockConsoleWarn.mockRestore();
    }
  });
});
