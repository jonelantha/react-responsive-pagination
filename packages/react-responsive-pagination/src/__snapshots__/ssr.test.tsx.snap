// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr renders the narrowest nav invisibly 1`] = `"<ul class="pagination justify-content-center" style="visibility:hidden"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;
